packages/account/src/providers/provider.ts

```diff
@@ -951,7 +950,7 @@ Supported fuel-core version: ${supportedVersion}.`
         const hasMissingOutputs =
           missingOutputVariables.length > 0 || missingOutputContractIds.length > 0;
         const request = allRequests[requestIdx];
-        if (hasMissingOutputs && request?.type === TransactionType.Script) {
+        if (hasMissingOutputs && request instanceof ScriptTransactionRequest) {
```

Comment:
@danielbate why using `instanceof` for this check? 

I have encountered situations where `instanceof` returned false when it should have returned true in similar contexts. Specifically, I've seen this issue when validating if a `TransactionRequest` is a `ScriptTransactionRequest`.

ref: https://github.com/FuelLabs/fuels-ts/pull/1702

Reply:
I'd wrongly assumed that `... === TransactionType.Script` didn't pass down types in block scope. Tbf @petertonysmith94 had also warned me.

Will change 👍🏻